snuba/datasets/configuration/entity_builder.py

```diff
@@ -40,9 +42,44 @@ def _build_entity_query_processors(
     ]
 
 
+def _parse_expressions(expressions_config: list[dict[str, Any]]) -> list[Expression]:
+    expressions_config = expressions_config.copy()
+    expressions: list[Expression] = []
+    for expression_config in expressions_config:
+        expression_name = expression_config["expression"]
+        if expression_name == "FunctionCall":
+            parameters = tuple(
+                _parse_expressions(expression_config["args"]["parameters"])
+            )
+            expression_config["args"]["parameters"] = parameters
+        expressions.append(
+            Expression.get_from_name(expression_name)(**expression_config["args"])
+        )
+    return expressions
+
+
+def _pre_process_column_mapper_config(sub_config: dict[str, Any]) -> dict[str, Any]:
+    # TODO: Will need to add support for ColumnToCurriedFunction when migrating sessions dataset to config
+    if sub_config["mapper"] == "ColumnToFunction":
+        sub_config["args"]["to_function_params"] = tuple(
+            _parse_expressions(sub_config["args"]["to_function_params"])
```

Comment:
There is actually already a parser that will convert a string into a correct expression. https://github.com/getsentry/snuba/blob/master/snuba/query/parser/expressions.py#L160

It's used in the CustomFunction processors (amongst other places).

Maybe a simpler solution here would be to represent the function as a string in the config, and then use the existing parser to convert it into an `Expression`. For example, `expression: "coalesce(IPv4NumToString(ip_address_v4), IPv6NumToString(ip_address_v6))"` .

Then `Expression` doesn't need to be a registered class and you don't need to reimplement an expression parser. This would also carry over to `ColumnToCurriedFunction` when we get there.

Reply:
☝️ yes let's do that. `Expression` classes are for the computer, not for humans. Humans should be able to type in actual expressions